base_tier_validation_helper/models/tier_definition_helper.py

```diff
@@ -0,0 +1,230 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+
+
+class TierDefinitionHelper(models.Model):
+    _name = 'tier.definition.helper'
+    _description = 'Tier Definition Helper'
+
+    @api.model
+    def _get_default_name(self):
+        return _('New')
+
+    name = fields.Char(
+        string='Description',
+        required=True,
+        default=lambda self: self._get_default_name(),
+        translate=True,
+    )
+    model_id = fields.Many2one(
+        comodel_name='ir.model',
+        string='Model',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    review_type = fields.Selection(
+        string='Validated by',
+        selection=[
+            ('individual', 'Specific user'),
+            ('group', 'Any user in a specific group.'),
+        ],
+        default='individual',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    definition_type = fields.Selection(
+        string='Definition',
+        selection=[
+            ('domain', 'Domain'),
+        ],
+        default='domain',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    auto_name = fields.Boolean(
+        string='Auto name',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='Auto name for tier definition',
+    )
+    notify_on_create = fields.Boolean(
+        string="Notify Reviewers on Creation",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="If set, all possible reviewers will be notified by email when "
+             "this definition is triggered."
+    )
+    has_comment = fields.Boolean(
+        string='Comment',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    approve_sequence = fields.Boolean(
+        string='Approve by sequence',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="Approval order by the specified sequence number",
+    )
+    state = fields.Selection(
+        string='Status',
+        selection=[
+            ('draft', 'Draft'),
+            ('done', 'Created'),
+            ('cancel', 'Deleted'),
+        ],
+        default='draft',
+        index=True,
+        copy=False,
+        track_visibility='onchange',
+        required=True,
+        readonly=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name='tier.definition.helper.line',
+        inverse_name='parent_id',
+        string='Lines',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.constrains('line_ids')
+    def _check_line_ids(self):
+        if self.state == 'done' and \
```

Comment:
As you use @api.multi, we need for loop on self.

Reply:
But why do we need this constraint? line_ids already readonly on state=='done' isn't it?

Reply:
We can use in view, create="0" and reduce state to only Draft -> Done.

Reply:
Fixed